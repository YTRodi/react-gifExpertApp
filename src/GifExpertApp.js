import React, { useState } from 'react'
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ( { defaultCategory = [] } ) => {

    // const categoriesDefault = [ 'Kimetsu no Yaiba', 'Shingeki no kyojin', 'Made in abyss' ];
    // const [ categories, setCategories ] = useState( categoriesDefault );

    // const [ categories, setCategories ] = useState( [ 'Tanjiro' ] );
    const [ categories, setCategories ] = useState( defaultCategory );

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories } />
            <hr />

            <ol>
                { 
                    // El key NO PUEDE ser el índice y TIENE QUE SER único.
                    categories.map( category =>

                        <GifGrid
                            key={ category }
                            category={ category } 
                        />
                        
                    )              
                }
            </ol>
        </>
    )
}

export default GifExpertApp
