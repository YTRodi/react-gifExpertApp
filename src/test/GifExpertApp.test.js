import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('Pruebas en el componente GifExpertApp', () => {
    
    test('Debe de mostrar el componente correctamente', () => {

        const wrapper = shallow( <GifExpertApp /> );

        expect( wrapper ).toMatchSnapshot();
        
    });

    test('Debe de mostrar una lista de categorias', () => {
        
        // Nota:
        // Como no hay una forma de testear los estados que poseen los componentes, le pasamos un
        // valor por defecto al mismo, para poder evaluarlo en los test.
        const categories = ['Shingeki no kyojin', 'One Piece'];
        const wrapper = shallow( <GifExpertApp defaultCategory={ categories } /> );
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( 'GifGrid' ).length ).toBe( categories.length );

    });

});