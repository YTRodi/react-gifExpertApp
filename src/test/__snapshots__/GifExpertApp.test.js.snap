// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifExpertApp Debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en el componente GifExpertApp Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Shingeki no kyojin"
      key="Shingeki no kyojin"
    />
    <GifGrid
      category="One Piece"
      key="One Piece"
    />
  </ol>
</Fragment>
`;
